datacats/tests/test_cli_pull.py

```diff
@@ -0,0 +1,23 @@
+from datacats.cli.pull import _retry_func
+from datacats.error import DatacatsError
+from unittest import TestCase
+
+
+def raise_an_error(_):
+    raise DatacatsError('Hi')
+
+
+class TestPullCli(TestCase):
+    def test_cli_pull_retry(self):
+        def count(*dummy, **_):
+            count.counter += 1
+        count.counter = 0
+
+        try:
+            _retry_func(raise_an_error, None, 5, count,
+                        'Error! We wanted this to happen')
+            self.fail('Exception was not raised.')
+        except DatacatsError:
+            pass
+        finally:
+            self.assertEqual(count.counter, 5)
```

Comment:
Hmm. does fail raise an exception? Might be better to move this assertEqual into the except statement. Also, let's check that the error that was raised has 'We wanted this to happen' in it, so our pass criteria is a little tighter.